src/lib/common/db/pool.js

```diff
@@ -1,55 +1,115 @@
-      localDataCenter: config.dc,
-      keyspace: config.keyspace,
-      credentials: { username: config.user, password: config.password },
-    });
-    await client.connect(); // Ensure the client is connected
-    isConnected = true;
-    console.log("Cassandra connection successful");
-    return true;
-  } catch (error) {
-    console.error("Cassandra connection error", error);
-    isConnected = false;
-    return false;
+  if (currEnv === "UAT") {
+    try {
+      clientUat = new cassandra.Client({
+        contactPoints: [poolConfig.pointOne, poolConfig.pointTwo],
+        localDataCenter: poolConfig.dc,
+        keyspace: poolConfig.keyspace,
+        credentials: {
+          username: poolConfig.user,
+          password: poolConfig.password,
+        },
+      });
+      await clientUat.connect(); // Ensure the client is connected
+      isConnectedUat = true;
+      console.log(
+        "Cassandra connection successful " + clientUat.options.localDataCenter
+      );
+      return true;
+    } catch (error) {
+      console.error("Cassandra connection error", error);
+      isConnectedUat = false;
+      return false;
+    }
+  } else if (currEnv === "PROD") {
+    try {
+      clientProd = new cassandra.Client({
+        contactPoints: [poolProdConfig.pointOne, poolProdConfig.pointTwo],
+        localDataCenter: poolProdConfig.dc,
+        keyspace: poolProdConfig.keyspace,
+        credentials: {
+          username: poolProdConfig.user,
+          password: poolProdConfig.password,
+        },
+      });
+      await clientProd.connect(); // Ensure the client is connected
+      isConnectedProd = true;
+      console.log(
+        "Cassandra connection successful " + clientProd.options.localDataCenter
+      );
+      return true;
+    } catch (error) {
+      console.error("Cassandra connection error", error);
+      isConnectedProd = false;
+      return false;
+    }
   }
 };
 
 const CloseCassandraConnection = async () => {
-  if (client && isConnected) {
+  if (clientUat && isConnectedUat) {
+    try {
+      await clientUat.shutdown();
+      isConnectedUat = false;
+      console.log("Cassandra connection closed for UAT");
+    } catch (error) {
+      console.error("Error closing Cassandra connection for UAT", error);
+    }
+  }
+  if (clientProd && isConnectedProd) {
     try {
-      await client.shutdown();
-      isConnected = false;
-      console.log("Cassandra connection closed");
+      await clientProd.shutdown();
+      isConnectedProd = false;
+      console.log("Cassandra connection closed for PROD");
     } catch (error) {
-      console.error("Error closing Cassandra connection", error);
+      console.error("Error closing Cassandra connection for PROD", error);
     }
   }
 };
 
 const SelectQuery = async (selectRowsQuery, env) => {
   try {
     await ConnectToCassandra(env);
-    const result = await client.execute(selectRowsQuery, [], { prepare: true });
```

Comment:
Use `ConnectToCassandra` to return the client and use it, instead of conditionally checking each time